### PR TITLE
Add equality operator for python Simplex tree

### PR DESCRIPTION
**Pull request by VincentRouvreau**

### CODE REVIEW

File: src/python/gudhi/simplex_tree.pyx [hunk 1]
```diff
@@ -639,3 +639,10 @@ cdef class SimplexTree:
             self.thisptr = <intptr_t>(ptr.collapse_edges(nb_iter))
             # Delete old pointer
             del ptr
+
+    def __eq__(self, other:SimplexTree):
```
Comment (mglisse) [inline src/python/gudhi/simplex_tree.pyx:643]: I wonder if that changes the minimum version of [pc]ython we require. (I think it is fine to require it)

File: src/python/gudhi/simplex_tree.pyx [hunk 2]
```diff
@@ -639,3 +639,10 @@ cdef class SimplexTree:
...
+        """Simplex tree equality operator using C++ depth first search operator==
```
Comment (mglisse) [inline src/python/gudhi/simplex_tree.pyx:644]: Is it really relevant to say it uses C++ and DFS? All that matters is that it is a structural equality.

File: src/python/gudhi/simplex_tree.pyx (hunk 1 above)
Comment (VincentRouvreau) [reply src/python/gudhi/simplex_tree.pyx:643]: Python typing involves python >= 3.5 ([source](https://docs.python.org/3/library/typing.html)) and cython >= 0.27 ([source](https://github.com/cython/cython/pull/3799/files)).

File: src/python/gudhi/simplex_tree.pyx (hunk 2 above)
Comment (mglisse) [reply src/python/gudhi/simplex_tree.pyx:644]:
```suggestion
        """Test for structural equality
```

File: src/python/gudhi/simplex_tree.pyx (hunk 1 above)
Comment (mglisse) [reply src/python/gudhi/simplex_tree.pyx:643]: Those requirements seem ok to me, should we document them?

**Review by mglisse — APPROVED** [review]

**Comment by VincentRouvreau** [comment]
@mglisse As you approved this PR, I will it at the same time as #552 will be approved as there is a clear dependency

**Pull request merged**